Source/JavaScriptCore/runtime/JSMicrotask.cpp

```diff
@@ -66,9 +66,16 @@ Ref<Microtask> createJSMicrotask(VM& vm, JSValue job, JSValue argument0, JSValue
 void runJSMicrotask(JSGlobalObject* globalObject, MicrotaskIdentifier identifier, JSValue job, JSValue argument0, JSValue argument1, JSValue argument2, JSValue argument3)
 {
     VM& vm = globalObject->vm();
+
     auto scope = DECLARE_CATCH_SCOPE(vm);
 
+    // If termination is issued, do not run microtasks. Otherwise, microtask should not care about exceptions.
+    if (vm.hasPendingTerminationException())
+        return;
+    scope.clearException();
+
     auto handlerCallData = JSC::getCallData(job);
+    scope.clearException();
```

Comment:
Shouldn't we also check for TerminationException here and return early if needed?

Reply:
Don't we always supply the `job` itself here? Like it can never come from userland code. Maybe we can `ASSERT` that there is no exception after `getCallData`?

Reply:
Done.